roffit

```diff
@@ -256,6 +256,33 @@ sub linkfile {
             # convert https://, http:// and ftp:// URLs to <a href> links
             $field =~ s/(^|\W)((https|http|ftp):\/\/[a-z0-9\-._~%:\/?\#\[\]\@!\$&'()*+,;=]+)/$1<a href=\"$2\">$2<\/a>/gi;
 
+            # remove punctuation characters at end of url
+            $field =~ s/(href=\")([^\"]*)(&quot;)/$1$2/g;
+            $field =~ s/(href=\"(http:\/\/|https:\/\/|ftp:\/\/)\")*([,.;()*])*(\">)/$1$4/g;
+
+            # null empty protocols
+            $field =~ s/href=\"(http:\/\/|https:\/\/|ftp:\/\/)\"/href=\"javascript:void(0)\"/g;
+
+            # add anchor and anchor links to special character options
+            my $specialcharacters = "`~!@\$%^*()-_=+{};:\'\\|,.?";
+            $field =~ s/(<a name=)\"-\"(><\/a>)(<span class=\"nroffip\">-)([$specialcharacters]|[\[\]])(,|<\/span>)/$1\"-$4\"$2$3$4/g;
+            $field =~ s/(<a class=\"emphasis\") href=\"#-\">-([$specialcharacters]|[\[\]])(,|<\/a>)/$1 href=\"#-$2\">-$2/g;
```

Comment:
I think these changes should rather be done as updates to `text2name()` and `do_encode()` instead of adding this extra stage of edits.

Reply:
Unable to achieve in subroutines, either anchor links were removed, all anchors (*name attribute*) removed, or no effect.

Reply:
> I think these changes should rather be done as updates to `text2name()` and `do_encode()` instead of adding this extra stage of edits.

With `text2name` and `do_encode` new things break. Some examples:
- name="socks5h://" (*should be name=\"socks5h\"*)
- name="AUNDERSCORE" (*shoud be name=\"A_UNDERSCORE*\")
- name=""--any-option" (*2 "" at start*).